onnxruntime/core/providers/cuda/tensor/gather_nd.cc

```diff
@@ -126,7 +126,7 @@ Status GatherNDBase::PrepareCompute(
 REGISTER_KERNEL_TYPED_GATHER_ND(int64_t, 1)
 #endif
 REGISTER_KERNEL_TYPED_GATHER_ND(int64_t, 13)
-REGISTER_KERNEL_VERSIONED_TYPED_GATHER_ND(int64_t, 12, 12)
+REGISTER_KERNEL_TYPED_GATHER_ND(int64_t, 12)
```

Comment:
What is the root cause? With this change, there is overlap between opset 12 and opset 13, which is supposed not.

Reply:
Thanks, I thought the combined key for kernel look is unique, will double check with Sherlock. The original code supports only AllIEEEFloatTensorTypes (i.e. float/double/MLFloat16) for the data type constraints of opset 12. When running mainz model, one GatherND is expecting int64_t. And since it's not part of the allowed data types, it doesn't have mapped CUDA kernel, thus forced to use CPU one. Thus the op before and after this op need to introduce extra memcpy(D2H/H2D,etc.).

Reply:
try 
.TypeConstraint("T", DataTypeImpl::AllTensorTypes()) ?

Reply:
or at least explicitly include int64, see op13 for reference

Reply:
ok .will try TypeConstraint("T", DataTypeImpl::AllTensorTypes()) instead